DBZ-1080 Document new EmbeddedEngine API (#244)
debezium/debezium.github.io: docs/embedded.asciidoc

Your application should always properly stop the engine to ensure graceful and complete shutdown and that each source record is sent to the application exactly one time. For example, do not rely upon shutting down the `ExecutorService`, since that interrupts the running threads. Although the `EmbeddedEngine` will indeed terminate when its thread is interrupted, the engine may not terminate cleanly, and when your application is restarted it may see some of the same source records that it had processed just prior to the shutdown.

[[advanced-consuming]]
== Advanced Record Consuming

For some use cases, such as when trying to write records in batches or against an async API, the functional interface described above may be challenging. In these situations, it may be easier to use the `io.debezium.embedded.EmbeddedEngine.ChangeConsumer` interface.

> This interface has single function the following signature:

Reviewer: ```suggestion
This interface has single function with the following signature:
```

[source,java,inden=0]
----
 /**
   * Handles a batch of records, calling the {@link RecordCommitter#markProcessed(SourceRecord)}
   * for each record and {@link RecordCommitter#markBatchFinished()} when this batch is finished.
   * @param records the records to be processed
   * @param committer the committer that indicates to the system that we are finished
   */
  void handleBatch(List<SourceRecord> records, RecordCommitter committer) throws InterruptedException;
----

> As mentioned in the javadoc, the RecordCommitter object is to be called for each record and once each batch is finished, but the RecordCommitter interface is threadsafe,

Reviewer: ```suggestion
As mentioned in the JavaDoc, the `RecordCommitter` object is to be called for each record and once each batch is finished.
The `RecordCommitter`contract guarantees thread-safety,
```

which allows for flexible processing of records.

> To use the `ChangeConsumer` api, the user must pass an implementation of the interface to the `notifying` API, as seen below:

Reviewer: ```suggestion
To use the `ChangeConsumer` API, you must pass an implementation of the interface to the `notifying()` API, as seen below:
```
